### PR TITLE
Add ratarmountcore

### PR DESCRIPTION
**Pull request by ap--**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ap--** [comment]
@mxmlnkn please confirm to be maintainer whenever you have time. There's absolutely no rush :christmas_tree: :relaxed:

**Comment by mxmlnkn** [comment]
I agree to be maintainer.

Merry Christmas to everyone involved.

**Comment by ap--** [comment]
@conda-forge/help-python, ready for review! 🎄

**Comment by mxmlnkn** [comment]
@conda-forge/help-python, ready for review!

File: recipes/ratarmountcore/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+test:
+  imports:
+    - ratarmountcore
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/mxmlnkn/ratarmount/ratarmountcore
```
Comment (dopplershift) [inline recipes/ratarmountcore/meta.yaml:34]:
```suggestion
  home: https://github.com/mxmlnkn/ratarmount/tree/master/core
```

File: recipes/ratarmountcore/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/ratarmountcore-{{ version }}.tar.gz
+  sha256: 320c87813aae4137881ab26641efb65d6b618ced002e2550787c1c2c2f851a53
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
```
Comment (dopplershift) [inline recipes/ratarmountcore/meta.yaml:14]: Looks like this is pure Python and could be noarch:
```suggestion
build:
  script: {{ PYTHON }} -m pip install . -vv
  number: 0
  noarch: python
```

File: recipes/ratarmountcore/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+requirements:
+  host:
+    - python
+    - pip
+  run:
+    - python
+    - dataclasses  # [py<37]
+    - importlib-metadata  # [py<38]
```
Comment (dopplershift) [inline recipes/ratarmountcore/meta.yaml:23]: For noarch (we can just unconditionally depend on the other packages):
```suggestion
    - python >=3.6
    - pip
  run:
    - python >=3.6
    - dataclasses
    - importlib-metadata
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just a few minor tweaks.

**Comment by ap--** [comment]
@dopplershift thank you so much for the review.
I committed your suggested changes.

For the future: is it generally recommended to permanently pull in dependencies for earlier python versions in favour of noarch?

**Comment by dopplershift** [comment]
> For the future: is it generally recommended to permanently pull in dependencies for earlier python versions in favour of noarch?

Yes, noarch drastically simplifies the build matrix and removes the need for many other rebuilds, like for new versions of Python.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**